File: qc-core/src/main/java/com/eager/questioncloud/core/domain/userquestion/UserQuestionJpaRepository.java
package com.eager.questioncloud.core.domain.userquestion;

import java.util.List;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserQuestionJpaRepository extends JpaRepository<UserQuestionEntity, Long> {
    Boolean existsByUserIdAndQuestionIdIn(Long userId, List<Long> questionIds);

    Boolean existsByUserIdAndQuestionId(Long userId, Long questionId);
}

File: qc-question/src/main/kotlin/com/eager/questioncloud/userquestion/infrastructure/repository/UserQuestionJpaRepository.kt
package com.eager.questioncloud.userquestion.infrastructure.repository

import com.eager.questioncloud.userquestion.infrastructure.entity.UserQuestionEntity
import org.springframework.data.jpa.repository.JpaRepository

interface UserQuestionJpaRepository : JpaRepository<UserQuestionEntity, Long> {
    fun existsByUserIdAndQuestionIdIn(userId: Long, questionIds: List<Long>): Boolean

    fun existsByUserIdAndQuestionId(userId: Long, questionId: Long): Boolean
}
